src/main.rs

```diff
@@ -46,32 +46,45 @@ impl PathTrie {
         join_with_parent: bool,
         lscolors: &LsColors,
         parent_path: PathBuf,
+        full_path: bool,
     ) {
         let normal_prefix = format!("{}│   ", prefix);
         let last_prefix = format!("{}    ", prefix);
 
         for (idx, (path, it)) in self.trie.iter().enumerate() {
             let current_path = parent_path.join(path);
             let style = ansi_style_for_path(&lscolors, &current_path);
-            let painted = style.paint(path.to_string_lossy());
+
+            let painted = match full_path {
+                false => style.paint(path.to_string_lossy()),
+                true => style.paint(current_path.to_string_lossy()),
+            };
 
             let contains_singleton_dir = it.contains_singleton_dir();
+            let dont_print = contains_singleton_dir && full_path;
```

Comment:
To avoid the double negative (`!dont_print`), can you call this variable `should_print`?

Also I had to think through why we want this condition in particular, so I'd love it if you also added a comment, something like this:

```suggestion
            // If this folder only contains a single dir, we skip printing it because it will be
            // picked up and printed on the next iteration. If this is a full path (even if it
            // contains more than one directory), we also want to skip printing, because the full
            // path will be printed all at once (see painted above), not part by part.
            let should_print = !contains_singleton_dir || !full_path;
```